### ISSUE TITLE
Use tower-service

### ISSUE THREAD

**Issue by emschwartz** [interledger-rs/interledger-rs#434]
We should consider getting rid of the `interledger-service::{IncomingService,OutgoingService}` traits and switching everything to [`tower-service`](https://github.com/tower-rs/tower/tree/master/tower-service).

When I started working on Interledger.rs, [`tower`](https://github.com/tower-rs/tower):
- still seemed like a work in progress
- didn't have any middleware I needed
- would have added more generics to the stack of services comprising Interledger.rs

Now, the `tower` ecosystem is farther along and we should consider switching. That said, it's probably worth waiting until [`v0.3.x`](https://github.com/tower-rs/tower/tree/v0.3.x) is released and until we switch to `std::Future`(#142).

**Comment by gakonst** [comment]
I agree that we should tackle this as soon as we move to futures 0.3 and while generally transitioning to async/await.
